Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterLauncher.cs

```diff
@@ -203,12 +203,16 @@ public sealed class DebugAdapterLauncher : IAdapterLauncher {
             launchJson.DebugStdLib = debugService.DebugStdLib;
             launchJson.ShowReturnValue = debugService.ShowFunctionReturnValue;
 
-            try {
-                var adapterLaunchInfoJson = JObject.Parse(adapterLaunchInfo.LaunchJson);
-                AddVariablePresentationOptions(adapterLaunchInfoJson, launchJson);
-            } catch (JsonReaderException) {
-                // this should never happen
-                Debug.Fail("adapterLaunchInfo is not valid json");
+            // adapterLaunchInfo.LaunchJson can be null when attaching to a remote process
+            if (adapterLaunchInfo.LaunchJson != null) {
```

Comment:
so what is AddDebuggerOptions for?

Reply:
so, if adapterLanguageInfo.launchJson is null, when this is called (https://github.com/microsoft/PTVS/blob/b5e38ce254348acce692b4cbe05c449df2c28c2b/Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterLauncher.cs#L71), what do we expect?

Reply:
https://github.com/microsoft/PTVS/blob/b5e38ce254348acce692b4cbe05c449df2c28c2b/Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugInfo.cs#L25-L105

Note that there are two different debug info classes there - one for launch, and one for attach. But they have a common base class, and `VariablePresentation` is there. So it seems like it should be available regardless of how the session gets started.

Reply:
Thanks, makes sense. I guess what confuses me is the two different ways that you can get these options. The one I used in https://github.com/microsoft/PTVS/pull/6337/files seems to be the `PythonToolsService`, and those options get packed into the adapterLaunchInfo.LaunchJson...but these other options are being read from the global `IPythonDebugOptionsService`.

It doesn't make sense to me to have two different paths to the same options. Should I just always be using the global one?

Reply:
It seems that I can just add some fields to the `IPythonDebugOptionsService` to expose the variable presentation stuff that already exists on the `DebuggerOptions` class. But this means that lots of the other code I added in the previous PR can go away, if we're just always going to be reading these options from the global options service.

Reply:
Hm, good point. This one is really meant for properties that are project-specific; for global properties, I guess it can just read it directly, since there's no ambiguity over which project it comes from.

@karthiknadig might have some thoughts about this.

Reply:
The two ways were there to handle project specific and global option. It should be fine to read non-project specific properties directly from global location. 

PS: This part is also a bit messy because of the way we had to handle `ptvsd4` vs `debugpy` at the same time.

Reply:
ok thank you everyone, I have a clear path now. Global makes sense to me too, since these options are not project specific. And since they are moving to global in vscode too, I think that's a good approach.